src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc

```diff
@@ -1174,24 +1107,22 @@ void FiniteDiffGroundTempsModel::evaluateSoilRhoCp(ObjexxFCL::Optional<int const
     auto &thisCell = cellArray(cell);
 
     //'set some temperatures here for generalization -- these could be set in the input file
-    frzAllIce = -0.5;
-    frzIceTrans = -0.4;
-    frzLiqTrans = -0.1;
-    frzAllLiq = 0.0;
+    Real64 frzAllIce = -0.5;
+    Real64 frzIceTrans = -0.4;
+    Real64 frzLiqTrans = -0.1;
+    Real64 frzAllLiq = 0.0;
 
     //'calculate this cell's new Cp value based on the cell temperature
     if (thisCell.temperature >= frzAllLiq) {
         rhoCP_soil = rhoCp_soil_liq_1;
     } else if (thisCell.temperature <= frzAllIce) {
         rhoCP_soil = rhoCP_soil_ice;
-    } else if ((thisCell.temperature < frzAllLiq) && (thisCell.temperature > frzLiqTrans)) {
+    } else if (thisCell.temperature > frzLiqTrans) {
         rhoCP_soil = rhoCp_soil_liq_1 + (rhoCP_soil_transient - rhoCP_soil_liq) / (frzAllLiq - frzLiqTrans) * (frzAllLiq - thisCell.temperature);
-    } else if ((thisCell.temperature <= frzLiqTrans) && (thisCell.temperature >= frzIceTrans)) {
+    } else if (thisCell.temperature >= frzIceTrans) {
         rhoCP_soil = rhoCP_soil_transient;
-    } else if ((thisCell.temperature < frzIceTrans) && (thisCell.temperature > frzAllIce)) {
-        rhoCP_soil = rhoCP_soil_ice + (rhoCP_soil_transient - rhoCP_soil_ice) / (frzIceTrans - frzAllIce) * (thisCell.temperature - frzAllIce);
```

Comment:
[src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc:1191]:(style),[knownConditionTrueFalse],Condition 'thisCell.temperature>frzAllIce' is always true